### ISSUE TITLE
Should the predict function in README.md return inputs rather than outputs?

### ISSUE THREAD

**Issue by juliangall** [google/jax#7910]
The example in the readme file has:

```python
def predict(params, inputs):
  for W, b in params:
    outputs = jnp.dot(inputs, W) + b
    inputs = jnp.tanh(outputs)
  return outputs
```

The return value should surely be the **inputs** variable (not **outputs**), as this is the output of the final tanh activation in the loop.

**Comment by zhangqiaorjc** [comment]
Thanks for finding a typo. Would you like to contribute a PR to fix this?

**Comment by jakevdp** [comment]
Actually, I believe this is correct as written. The variable names are a bit confusing, but the intent is to return the result of the final dot product, not the result of the final `tanh` activation function.

**Comment by zhangqiaorjc** [comment]
@jakevdp  Oops..I read the snippet in isolation. Could we maybe add a comment at least? This is confusing on a cursory look...

**Comment by jakevdp** [comment]
We've discussed this previously and ended up renaming variables, and talked about adding a comment but never got consensus on what to add (a sampling of similar issues with relevant discussions: #31, #119, #4048, #4560, #5230, #5236, #7888)

**Comment by juliangall** [comment]
Thanks everyone. A comment will be helpful. My preference would be to have the code do what is necessary to express its intent, and not to "throw away" the final ```tanh``` activation. However, I am sure others will feel differently.

**Comment by jakevdp** [comment]
Thanks! I'm going to close this, since I think the issue should be addressed by #7915
